userspace/libsinsp/parsers.cpp

```diff
@@ -1612,6 +1614,38 @@ void sinsp_parser::parse_clone_exit(sinsp_evt *evt)
 	return;
 }
 
+void sinsp_parser::parse_execve_enter(sinsp_evt *evt)
+{
+	store_event(evt);
+
+	if(!evt->m_tinfo)
+	{
+		// Should be impossible
+		ASSERT(false);
+
+		return;
+	}
+
+	// Find the main thread for this pid and note that this tid
+	// was an an exec enter event. When parsing the exit event,
+	// the threadinfo for the tid will be removed, as it no longer
+	// exists.
+	sinsp_threadinfo* main_thread = m_inspector->get_thread_ref(evt->m_tinfo->m_pid, true, true).get();
+
+	if(!main_thread)
+	{
+		return;
+	}
+
+	// Only set the exec enter tid for non-main threads
+	if(main_thread->m_tid == evt->m_tinfo->m_tid)
```

Comment:
Can't we always store the exec_enter_tid here (even for the main thread), and then let the `parse_execve_exit` manage it for any thread of the process?
Instead of doing this workaround: https://github.com/falcosecurity/libs/pull/336/files#diff-f1f5adc2cc836ddaf885f86ded61c940d5c957781ea3ef5cef7f4c6f407054f7R1434